### PR TITLE
Filterable markdown

### PR DESCRIPTION
**Pull request by mAAdhaTTah**
Exposes filters to allow customization of the `CommonMarkConverter`.

### CODE REVIEW

File: markdown-on-save.php [hunk 1]
```diff
@@ -31,6 +32,11 @@ public function __construct() {
   }
 
   public function init() {
+    $this->markdown = new CommonMarkConverter(
+        apply_filters('cws_markdown_config', array()),
```
Comment (swalkinshaw) [inline markdown-on-save.php]: Small nitpick: `[]` instead of `array()`

**Comment by swalkinshaw** [comment]
This is awesome :+1: I recently made a branch where I hardcoded this just to experiment: https://github.com/roots/markdown-on-save/tree/markdown-tables-support

File: markdown-on-save.php (hunk 1 above)
Comment (mAAdhaTTah) [inline markdown-on-save.php]: :+1:

**Comment by mAAdhaTTah** [comment]
Yeah, I really need a footnotes implementation, but I might have to do that myself.

**Pull request merged**